core/src/main/java/com/novoda/noplayer/internal/exoplayer/mediasource/MediaSourceFactory.java

```diff
@@ -1,60 +1,74 @@
 package com.novoda.noplayer.internal.exoplayer.mediasource;
 
+import android.content.Context;
 import android.net.Uri;
 import android.os.Handler;
 
 import com.google.android.exoplayer2.extractor.DefaultExtractorsFactory;
-import com.google.android.exoplayer2.source.AdaptiveMediaSourceEventListener;
 import com.google.android.exoplayer2.source.ExtractorMediaSource;
 import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.source.MediaSourceEventListener;
 import com.google.android.exoplayer2.source.dash.DashMediaSource;
 import com.google.android.exoplayer2.source.dash.DefaultDashChunkSource;
 import com.google.android.exoplayer2.source.hls.HlsMediaSource;
-import com.google.android.exoplayer2.upstream.DataSource;
-import com.novoda.noplayer.ContentType;
+import com.google.android.exoplayer2.upstream.DefaultBandwidthMeter;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.novoda.noplayer.Options;
 
 public class MediaSourceFactory {
 
-    private final DataSource.Factory mediaDataSourceFactory;
+    private final Context context;
     private final Handler handler;
 
-    public MediaSourceFactory(DataSource.Factory mediaDataSourceFactory, Handler handler) {
-        this.mediaDataSourceFactory = mediaDataSourceFactory;
+    public MediaSourceFactory(Context context, Handler handler) {
+        this.context = context;
         this.handler = handler;
     }
 
-    public MediaSource create(ContentType contentType,
+    public MediaSource create(Options options,
                               Uri uri,
-                              ExtractorMediaSource.EventListener eventListener,
-                              AdaptiveMediaSourceEventListener mediaSourceEventListener) {
-        switch (contentType) {
+                              MediaSourceEventListener mediaSourceEventListener) {
+        DefaultBandwidthMeter bandwidthMeter = new DefaultBandwidthMeter.Builder()
+                .setInitialBitrateEstimate(options.maxInitialBitrate())
+                .build();
+        DefaultDataSourceFactory defaultDataSourceFactory = new DefaultDataSourceFactory(context, "user-agent", bandwidthMeter);
```

Comment:
Delayed creation of this and the `BandwidthMeter` so that we can make use of the `Options` to create the `BandwidthMeter` rather than passing a param as part of the `Factories` and `Creators`.